test/integration/lerna-run.test.js

```diff
@@ -0,0 +1,20 @@
+import execa from "execa";
+import initFixture from "../helpers/initFixture";
+import { LERNA_BIN } from "../helpers/constants";
+
+const installInDir = (cwd) =>
+  execa("npm", ["install", "--cache-min=99999"], { cwd });
+
+describe("lerna run", () => {
+
+  test.concurrent("can run script in packages", () => {
+    return initFixture("RunCommand/basic").then((cwd) => {
+      return Promise.resolve()
+        .then(() => installInDir(cwd))
+        .then(() => execa(LERNA_BIN, ["run", "my-script", "--scope=package-1"], { cwd }))
```

Comment:
The snapshots need to be "stable", that is, they can't have the temp directory in them, because that changes every run. To pass the `--silent` flag to `npm` (which omits the logging that contains the unstable tmpdir), add `--`, `--silent` to the args here. It's how I got the bootstrap integration `lerna run test` to have stable snapshots.

I agree it's a bit weird, especially how we handle the "extra args" (the ones that follow `--` aren't parsed by yargs or meow, but passed on to `npm run`, _as args to `npm`, not the script_.